src/test/java/org/jitsi/meet/test/pageobjects/web/KnockingParticipantList.java

```diff
@@ -0,0 +1,177 @@
+     * @return list of knocking participants.
+     */
+    public List<Participant> getParticipants()
+    {
+        List<Participant> participants = new ArrayList<>();
+
+        try
+        {
+            TestUtils.waitForDisplayedElementByID(participant.getDriver(), KNOCKING_PARTICIPANT_LIST_ID, 3);
+        }
+        catch(TimeoutException ex)
+        {
+            // if the list is missing return empty list of participants
+            Logger.getGlobal().log(Level.WARNING, "No knocking participant list ");
+            return participants;
+        }
+
+        List<WebElement> listElements = participant.getDriver().findElements(By.xpath(PARTICIPANTS_XPATH));
+
+        listElements.forEach(el -> {
+            String avatarUrl = el.findElement(ByTestId.testId(PARTICIPANT_AVATAR_TEST_ID)).getAttribute("src");
+            String name = el.findElement(ByTestId.testId(PARTICIPANT_NAME_TEST_ID)).getText();
+            String email = null;
+                try
+                {
+                    email = el.findElement(By.id(PARTICIPANT_EMAIL_TEST_ID)).getText();
+                }
+                catch(NoSuchElementException e)
+                {
+                    // email is optional and may be missing
+                }
+            participants.add(new Participant(avatarUrl, name, email));
+        });
+
+        return participants;
+    }
+
+    /**
+     * Presentation for the participants in the knocking participants list.
+     */
+    public class Participant
+    {
+        private final String avatarUrl;
+        private final String name;
+        private final String email;
+
+        public Participant(String avatarUrl, String name, String email)
+        {
+            this.avatarUrl = avatarUrl;
+            this.name = name;
+            this.email = email;
+        }
+
+        public String getAvatarUrl()
+        {
+            return avatarUrl;
+        }
+
+        public String getName()
+        {
+            return name;
+        }
+
+        public String getEmail()
+        {
+            return email;
+        }
+
+        /**
+         * Allow participant to join meeting.
+         */
+        public void allow()
+        {
+            clickButton(PARTICIPANT_ALLOW_TEST_ID);
+        }
+
+        /**
+         * Denies participant access.
+         */
+        public void reject()
+        {
+            clickButton(PARTICIPANT_REJECT_TEST_ID);
+        }
+
+        /**
+         * Clicks the action button for the current participant.
+         * @param buttonId the button Id to click.
+         */
+        private void clickButton(String buttonId)
+        {
+            List<WebElement> listElements = participant.getDriver().findElements(By.xpath(PARTICIPANTS_XPATH));
+            WebElement participantElement = listElements.stream().filter(participant -> {
+                String name = participant.findElement(ByTestId.testId(PARTICIPANT_NAME_TEST_ID)).getText();
```

Comment:
probably it's doable with a selector, but not a big deal since it's contained in this method